.ci/aarch64/install_rom_aarch64.sh

```diff
@@ -0,0 +1,145 @@
+EDK2_PLAT_REPO="https://github.com/tianocore/edk2-platforms.git"
+ACPICA="https://github.com/acpica/acpica.git"
+
+
+#build toolchain
+TOOLCHAIN_VERSION="10.2-2020.11"
+TOOLCHAIN_ARCHIVE_PREFIX="gcc-arm-${TOOLCHAIN_VERSION}-aarch64-aarch64-none-elf"
+TOOLCHAIN_ARCHIVE="${TOOLCHAIN_ARCHIVE_PREFIX}.tar.xz"
+TOOLCHAIN_PREFIX="${TOOLCHAIN_ARCHIVE_PREFIX}/bin/aarch64-none-elf-"
+TOOLCHAIN_SOURCE_URL="https://developer.arm.com/-/media/Files/downloads/gnu-a/${TOOLCHAIN_VERSION}/binrel/${TOOLCHAIN_ARCHIVE}"
+
+export WORKSPACE=$(mktemp -d)
+
+QEMU_EFI_BUILD_PATH="${WORKSPACE}/Build/ArmVirtQemu-AARCH64/RELEASE_GCC5/FV/QEMU_EFI.fd"
+
+PREFIX="${PREFIX:-/usr}"
+INSTALL_PATH="${DESTDIR:-}${PREFIX}/share/kata-containers/"
+
+EFI_NAME="QEMU_EFI.fd"
+EFI_DEFAULT_DIR="/usr/share/qemu-efi-aarch64"
+EFI_DEFAULT_PATH="${EFI_DEFAULT_DIR}/${EFI_NAME}"
+
+FLASH0_NAME="kata-flash0.img"
+FLASH1_NAME="kata-flash1.img"
+
+HOW_TO_CROSS_BUILD="https://developer.arm.com/tools-and-software/open-source-software/firmware/edkii-uefi-firmware/building-edkii-uefi-firmware-for-arm-platforms"
+
+arch=$(uname -m)
+
+build_uefi()
+{
+	cd "${WORKSPACE}"
+	git clone "${EDK2_REPO}"
+	git clone "${EDK2_PLAT_REPO}"
+	git clone "${ACPICA}"
+
+
+	sudo apt install -y python python3 python3-distutils uuid-dev build-essential bison flex
+
+	mkdir toolchain
+	cd toolchain/
+	curl -LO "${TOOLCHAIN_SOURCE_URL}" && tar -xf "${TOOLCHAIN_ARCHIVE}"
+	cd ..
+
+	make -C acpica/
+
+	export GCC5_AARCH64_PREFIX="${WORKSPACE}/toolchain/${TOOLCHAIN_PREFIX}"
+	export PACKAGES_PATH=$WORKSPACE/edk2:$WORKSPACE/edk2-platforms
+	export IASL_PREFIX=$WORKSPACE/acpica/generate/unix/bin/
+
+	export PYTHON_COMMAND=/usr/bin/python3
+
+	cd edk2 && git submodule update --init && cd -
+	source edk2/edksetup.sh
+	make -C edk2/BaseTools
+
+	build -a AARCH64 -t GCC5 -p edk2/ArmVirtPkg/ArmVirtQemu.dsc -b RELEASE
+	echo "Info: build uefi successfully"
+}
+
+prepare_default_uefi()
+{
+	if [ "$ID" == "ubuntu" ] || [ "$ID" == "debian" ]; then
+		sudo apt remove -y qemu-efi-aarch64
+		sudo apt install -y qemu-efi-aarch64
+	else
+		local efi_url="https://releases.linaro.org/components/kernel/uefi-linaro/latest/release/qemu64/QEMU_EFI.fd"
+		sudo mkdir -p "${EFI_DEFAULT_DIR}"
+		cd "${EFI_DEFAULT_DIR}"
+		sudo curl -LO "$efi_url"
+		cd -
+	fi
+}
+
+prepare_uefi_flash() {
+	cd "${WORKSPACE}"
+
+	dd if=/dev/zero of=${FLASH0_NAME} bs=1M count=64
+	dd if="$1" of=${FLASH0_NAME} conv=notrunc
+	dd if=/dev/zero of=${FLASH1_NAME} bs=1M count=64
+}
+
+install_uefi_flash()
+{
+	[ -z "$1" -o -z "$2" ] && clean_up_and_die "fail to install uefi flash for lack of input"
+	[ -d "${INSTALL_PATH}" ] || mkdir -p ${INSTALL_PATH}
+	sudo install --mode 0544 -D "$1" "${INSTALL_PATH}/${FLASH0_NAME}"
+	sudo install --mode 0544 -D "$2" "${INSTALL_PATH}/${FLASH1_NAME}"
+}
+
+clean_up_and_die()
+{
+	clean_up
+	echo "ERROR: $*" >&2
+	exit 1
+}
+
+clean_up()
+{
+	sudo rm -rf "${WORKSPACE}"
+}
+
+main()
+{
+	[ $USER == "root" ] && echo "run this script as root"
```

Comment:
This variable can be modified so it's not as safe as checking `id -u`

Reply:
OK.